File: seed/php-model/nullable-allof-extends/src/RootObject.php
<?php

namespace Seed;

use Seed\Core\Json\JsonSerializableType;
use Seed\Traits\NormalObject;
use Seed\Traits\NullableObject;

/**
 * Object inheriting from a nullable schema via allOf.
 */
class RootObject extends JsonSerializableType
{
    use NormalObject;
    use NullableObject;


    /**
     * @param array{
     *   normalField?: ?string,
     *   nullableField?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->normalField = $values['normalField'] ?? null;
        $this->nullableField = $values['nullableField'] ?? null;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: seed/ruby-sdk-v2/nullable-allof-extends/lib/seed/types/root_object.rb
# frozen_string_literal: true

module Seed
  module Types
    # Object inheriting from a nullable schema via allOf.
    class RootObject < Internal::Types::Model
      field :normal_field, -> { String }, optional: true, nullable: false, api_name: "normalField"

      field :nullable_field, -> { String }, optional: true, nullable: false, api_name: "nullableField"
    end
  end
end
